File: src/main/java/com/arnas/klatrebackend/features/climbingsessions/ClimbingSessionRepository.java
package com.arnas.klatrebackend.features.climbingsessions;

import org.springframework.lang.Nullable;

import java.util.List;

public interface ClimbingSessionRepository {
    @Nullable ClimbingSession getClimbingSessionById(long sessionId);
    @Nullable ClimbingSession getActiveSession(long groupId, long userId);
    List<ClimbingSession> getPastSessions(long groupId, long userId);
    long openActiveSession(long userId, long groupId, long placeId);
    int setSessionAsInactive(long activeSessionId);
    long uploadClimbingSession(ClimbingSessionDTO climbingSession);
    int deleteClimbingSession(long climbingSessionId);
    List<RouteAttempt> getRouteAttemptsBySessionId(long sessionId);
    int updateRouteAttempt(RouteAttempt routeAttempt);
    int deleteRouteAttempt(long routeAttemptId);
    RouteAttempt addRouteAttemptToActiveSession(long activeSessionId, RouteAttemptDTO routeAttempt);
    RouteAttempt getRouteAttemptById(long id);
}

File: src/main/kotlin/com/arnas/klatrebackend/features/climbingsessions/ClimbingSessionRepository.kt
package com.arnas.klatrebackend.features.climbingsessions

interface ClimbingSessionRepository {
    fun getClimbingSessionById(sessionId: Long): ClimbingSession?
    fun getActiveSession(groupId: Long, userId: Long): ClimbingSession?
    fun getPastSessions(groupId: Long, userId: Long): List<ClimbingSession>
    fun openActiveSession(userId: Long, groupId: Long, placeId: Long): Long
    fun setSessionAsInactive(activeSessionId: Long): Int
    fun uploadClimbingSession(climbingSession: ClimbingSessionDTO): Long
    fun deleteClimbingSession(climbingSessionId: Long): Int
    fun getRouteAttemptsBySessionId(sessionId: Long): List<RouteAttempt>
    fun updateRouteAttempt(routeAttempt: RouteAttempt): Int
    fun deleteRouteAttempt(routeAttemptId: Long): Int
    fun addRouteAttemptToActiveSession(activeSessionId: Long, routeAttempt: RouteAttemptDTO): RouteAttempt
    fun getRouteAttemptById(id: Long): RouteAttempt
}

